apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -18,44 +18,50 @@
 
 defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
   alias Astarte.Core.CQLUtils
-  alias Astarte.Core.Device
+  alias Astarte.Core.Device, as: CoreDevice
   alias Astarte.Core.InterfaceDescriptor
   alias Astarte.Core.Mapping
   alias Astarte.DataUpdaterPlant.Config
+  alias Astarte.DataUpdaterPlant.DataUpdater.SimpleTrigger
+  alias Astarte.DataUpdaterPlant.DataUpdater.Device
+  alias Astarte.DataUpdaterPlant.DataUpdater.Endpoint
+  alias Astarte.DataUpdaterPlant.DataUpdater.IndividualProperty
+  alias Astarte.DataUpdaterPlant.DataUpdater.KvStore
   alias CQEx.Query, as: DatabaseQuery
-  alias CQEx.Result, as: DatabaseResult
+  alias Astarte.DataUpdaterPlant.Repo
+  import Ecto.Query
   require Logger
 
-  def query_simple_triggers!(db_client, object_id, object_type_int) do
-    simple_triggers_statement = """
-    SELECT simple_trigger_id, parent_trigger_id, trigger_data, trigger_target
-    FROM simple_triggers
-    WHERE object_id=:object_id AND object_type=:object_type_int
-    """
+  def query_simple_triggers!(realm, object_id, object_type_int) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
-    simple_triggers_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(simple_triggers_statement)
-      |> DatabaseQuery.put(:object_id, object_id)
-      |> DatabaseQuery.put(:object_type_int, object_type_int)
+    query =
+      SimpleTrigger
+      |> where(object_id: ^object_id, object_type: ^object_type_int)
+      |> put_query_prefix(keyspace_name)
 
-    DatabaseQuery.call!(db_client, simple_triggers_query)
+    Repo.all(query)
   end
 
-  def query_all_endpoint_paths!(db_client, device_id, interface_descriptor, endpoint_id) do
-    all_paths_statement = """
-    SELECT path FROM #{interface_descriptor.storage}
-    WHERE device_id=:device_id AND interface_id=:interface_id AND endpoint_id=:endpoint_id
-    """
+  def all_device_owned_property_endpoint_paths!(
+        realm,
+        device_id,
+        interface_descriptor,
+        endpoint_id
+      ) do
+    %InterfaceDescriptor{interface_id: interface_id, storage: storage} = interface_descriptor
 
-    all_paths_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(all_paths_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:interface_id, interface_descriptor.interface_id)
-      |> DatabaseQuery.put(:endpoint_id, endpoint_id)
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
+
+    q =
+      from(storage)
+      |> select([:path])
```

Comment:
To keep things simple we can probably skip this select.
Otherwise, we could return directly a list of paths, which is what the caller is interested in.

```elixir
from(storage)
|> select([s], s.path)
|> ...
```